scripts/install-nix-from-closure.sh

```diff
@@ -1,181 +1,700 @@
+#
+###############################
+#
+# This is the best balance of code simplicity and code readability found so far.
+# This produces output in color:
+#
+###############################
+#
+#./install: Notice:
+#
+#    Install executed for ROOT.
+#
+#    Doing classic Linux package manager mode.
+#    In Nix this mode is called: single-user mode for root.
+#
+#    Nix can do multi-user mode, and manage package trees for users
+#    independently.
+#
+###############################
+#
+true # so {...} body has some code, shell will not permit otherwise
+}
+###############################
+###  Setup environment
+###############################
+{
+# Set the character collating sequence to be numeric ASCII/C standard.
+# Set the character set to be standard one-byte ASCII.
+# Set default umask; to be non-restrictive and friendly to others.
+#
+LC_COLLATE=C
+LANG=C
+umask 022
+}
+###############################
+###  Main constants
+###############################
+{
+readonly dest="/nix"
+readonly self="$(dirname "$(realpath "$0")")"
+readonly nix="@nix@"
+readonly cacert="@cacert@"
+readonly appname="$0"
+}
+###############################
+###  CLI control constants
+###############################
+{
+# If file descriptor is associated with a terminal - output colors
+if test -t ; then
+    # If terminal reports a terminfo/termcap DB of colors available - use DB.
+    # Else - use literal color codes.
+    if tput colors > /dev/null 2>&1 ; then
+        # use tput and terminfo DB
+        readonly red=$(tput setaf 1)
+        readonly green=$(tput setaf 2)
+        readonly yellow=$(tput setaf 3)
+        readonly blue=$(tput setaf 4)
+        readonly bold=$(tput smso)
+        readonly reset=$(tput sgr0) # Reset to default output
+    else
+        # tput is not present on some systems (Alpine Linux),
+        # this trick with printf allows to store, not 'codes' - literal symbols
+        readonly red=$(printf '\033[1;31m')
+        readonly green=$(printf '\033[1;32m')
+        readonly yellow=$(printf '\033[1;33m')
+        readonly blue=$(printf '\033[1;34m')
+        readonly bold=$(printf '\033[1m')
+        readonly reset=$(printf '\033[0;m') # Reset to default output
+    fi
+fi
+}
+###############################
+###  CLI output functions
+###############################
+{
+# Unified output function
+print() {
+    # Using `printf`, because it is more portable than `echo`.
+    #
+    if [ -z "$message" ]; then
+        message="$1"
+    fi
+    if [ -z "$color" ]; then
+        color="$reset"
+    fi
+    if [ -z "$prefix" ]; then
+        prefix='Info'
+    fi
+    # This line makes all prints in script
+    # Form of message:
+    # Application: Prefix: Body of message
+    printf '%s%s: %s: %s%s\n' "$color" "$appname" "$prefix" "$message" "$reset"
+
+    # Output is done, unset variables that could've been set by parent functions
+    unset color
+    unset prefix
+    unset message
+}
+
+notice() {
+    message="$1"
```

Comment:
Almost all your variable assignments use more quotes than needed.

Reply:
Yeah.

Since variables in variable assignment work differently then variables in commands, I adopted the universal style for consistency and additional robustness.

We can remove those, but then in writing it is one more point to remember about. And more complex style also adds confusion for people that try to understand it.

People get lost in Shell quoting labyrinths. Hell, sometimes - I do also.

Reply:
But to be proper - it is really not needed and can be removed.
It is a syntactic sugar for work and to ease code readability.

Reply:
You may be right, in some cases of complex quotation additional quotes here can be excessive.
But it is a minor issue here.
This quotes does not influence even the speed of script parser even a scooch, it runs as fast, they are free quotes in Haskell terms.
And the code works.